glusterapi/volume.py

```diff
@@ -1,132 +1,275 @@
+
+        self.volume_stop(vol_name)
+        return self.volume_start(vol_name, force)
 
-    def volume_delete(self, volname, force=False):
+    def volume_delete(self, vol_name):
         """
-        Start Gluster Volume
+        Start Gluster Volume.
 
         :param volname: (string) Volume Name
-        :param force: (bool) Stop Volume with Force
         :raises: GlusterAPIError or failure
         """
-        data = {
-            "force": force
-        }
-        self._handle_request(self._delete, 200,
-                             "/v1/volumes/%s" % volname, data=data)
+        vol_name = vol_name.strip()
+        if not vol_name:
+            raise GlusterApiInvalidInputs("Volume name cannot be empty")
+
+        return self._handle_request(self._delete, httplib.NO_CONTENT,
+                                    "/v1/volumes/%s" % vol_name, None)
 
-    def volume_set(self, volname, optname, optvalue):
+    def volume_set(self, vol_name, options=None,
+                   advance=False,
+                   experimental=False,
+                   deprecated=False):
         """
-        Start Gluster Volume
+        Start Gluster Volume.
 
-        :param volname: (string) Volume Name
-        :param force: (bool) Stop Volume with Force
-        :raises: GlusterAPIError or failure
+        :param vol_name: (string) Volume Name
+        :param options: (dict) options to set on volume
+        :param advance: (bool) advance flag to set options
+        :param experimental: (bool) experimental flag to set options
+        :param deprecated: (bool) deprecated flag to set options
+        :raises: GlusterApiError or GlusterApiInvalidInputs on failure
         """
-        pass
+        vol_name = vol_name.strip()
+        if not vol_name:
+            raise GlusterApiInvalidInputs("Volume name cannot be empty")
+
+        if options is None:
+            raise GlusterApiInvalidInputs("cannot set empty options")
+
+        voloptions = dict()
+        req = dict()
+        for key in options:
+            voloptions[key] = options[key]
+        req['options'] = voloptions
+        req['advanced'] = advance
+        req['experimental'] = experimental
+        req['deprecated'] = deprecated
+        return self._handle_request(self._post, httplib.OK,
+                                    "/v1/volumes/%s/options" % vol_name,
+                                    json.dumps(req))
 
-    def volume_reset(self, volname, optnames=[]):
+    def volume_reset(self, vol_name, options=None):
         """
-        Start Gluster Volume
+        Start Gluster Volume.
 
         :param volname: (string) Volume Name
-        :param force: (bool) Stop Volume with Force
+        :param options: (dict) options to set on volume
         :raises: GlusterAPIError or failure
         """
+        vol_name = vol_name.strip()
+        if not vol_name:
+            raise GlusterApiInvalidInputs("Volume name cannot be empty")
+
```

Comment:
Maybe you can leave a comment here with a TODO that you'll complete this once the PR is merged to GD2.